File: HealthTracker/Model/HealthKitManager.m
//
//  HealthKitManager.m
//  HealthTracker
//
//  Created by John on 7/18/17.
//  Copyright © 2017 Bruce32. All rights reserved.
//

#import "HealthKitManager.h"

@interface HealthKitManager	()

@property (nonatomic, strong) HKHealthStore *healthStore;

@end


@implementation HealthKitManager

- (id) init
{
	self = [super init];
	self.healthStore = [[HKHealthStore alloc] init];
	
	[self.healthStore requestAuthorizationToShareTypes:NULL readTypes:[HealthKitManager dataTypesToRead] completion: ^(BOOL success, NSError * error)
	 {
	 }];
	
	return self;
}

+ (NSSet<HKObjectType*> *) dataTypesToRead
{
	return [[NSSet alloc] initWithArray: @[
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryFatTotal],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryFatPolyunsaturated],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryFatMonounsaturated],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryFatSaturated],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryCholesterol],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietarySodium],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryCarbohydrates],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryFiber],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietarySugar],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryEnergyConsumed],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryProtein],
			
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryVitaminA],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryVitaminB6],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryVitaminB12],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryVitaminC],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryVitaminD],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryVitaminE],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryVitaminK],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryCalcium],
			
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryIron],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryThiamin],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryRiboflavin],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryNiacin],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryFolate],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryBiotin],
			
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryPantothenicAcid],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryPhosphorus],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryIodine],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryMagnesium],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryZinc],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietarySelenium],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryCopper],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryManganese],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryChromium],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryMolybdenum],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryChloride],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryPotassium],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryCaffeine],
			[HKQuantityType quantityTypeForIdentifier: HKQuantityTypeIdentifierDietaryWater] ]];
}

@end

File: HealthTracker/Model/HealthKitManager.swift
//
//  HealthKitManager.swift
//  HealthTracker
//
//  Created by John on 7/25/17.
//  Copyright © 2017 Bruce32. All rights reserved.
//

import Foundation
import HealthKit

class HealthKitManager {

	private var healthStore: HKHealthStore
	private let readTypes: Set<HKObjectType>

	init() {

		healthStore = HKHealthStore()
		readTypes = HealthKitManager.getReadTypes()
		
		healthStore.requestAuthorization(toShare: nil, read: readTypes) { (_, _) in }
	}

	private static func getReadTypes() -> Set<HKObjectType> {

		return [HKQuantityType.quantityType(forIdentifier: .dietaryCarbohydrates)!,
		        HKQuantityType.quantityType(forIdentifier: .dietaryCholesterol)!,
		        HKQuantityType.quantityType(forIdentifier: .dietaryEnergyConsumed)!,
		        HKQuantityType.quantityType(forIdentifier: .dietaryFatSaturated)!,
		        HKQuantityType.quantityType(forIdentifier: .dietaryFatTotal)!,
		        HKQuantityType.quantityType(forIdentifier: .dietaryFiber)!,
		        HKQuantityType.quantityType(forIdentifier: .dietaryProtein)!,
		        HKQuantityType.quantityType(forIdentifier: .dietarySodium)!,
		        HKQuantityType.quantityType(forIdentifier: .dietarySugar)!,
		        HKQuantityType.quantityType(forIdentifier: .dietaryVitaminA)!,
		        HKQuantityType.quantityType(forIdentifier: .dietaryVitaminB12)!,
		        HKQuantityType.quantityType(forIdentifier: .dietaryVitaminB6)!,
		        HKQuantityType.quantityType(forIdentifier: .dietaryVitaminC)!,
		        HKQuantityType.quantityType(forIdentifier: .dietaryVitaminD)!,
		        HKQuantityType.quantityType(forIdentifier: .dietaryVitaminE)!,
		        HKQuantityType.quantityType(forIdentifier: .dietaryVitaminE)!]
	}
}
